include/albatross/src/tune/greedy_tuner.hpp

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (C) 2021 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+#ifndef ALBATROSS_SRC_TUNE_GREEDY_TUNER_HPP_
+#define ALBATROSS_SRC_TUNE_GREEDY_TUNER_HPP_
+
+namespace albatross {
+
+/*
+ * The greedy_tune method is an alternative to the other tuning approaches
+ * in albatross.  It was designed for coarse tuning of computationally
+ * complex objective functions.  The method uses multi-threading to drive
+ * a relatively simple tuning approach which works by:
+ *
+ *   - Picking one of the parameters.
+ *   - Using the range defined by the prior to produce new guesses.
+ *   - Evaluating the objective at all the guesses (one guess per thread).
+ *   - Picking the best parameter value.
+ *   - Repeating for the next parameter.
+ *
+ * There are almost certainly no global convergence guarantees with
+ * this algorithm, but in practice it seems to do a good job of coarsely
+ * tuning the objective.  One strategy would be to start with an iteration
+ * of this greedy approach and follow up with one of the methods in tune.hpp
+ */
+
+namespace details {
+
+inline std::vector<double> query_ratios(std::size_t n) {
+  const double log_min = -5.;
```

Comment:
Where do these numbers come from? 1e-5 to 1e-.1

Reply:
This is a rudimentary way of picking which values should get attempted depending on how many threads you have at your disposal.  Perhaps this helps, here is a diagram of the perturbations which will get attempted as a function of `n`:

![image](https://user-images.githubusercontent.com/514053/120502927-b2030900-c377-11eb-831d-fce5f88d1d27.png)

basically, if `n` is one we only try a small perturbation from the current values.  When it's two, we try a small perturbation and then something near the upper bound, and as n increases we fill in with more attempts in the middle.  The log scale is to favor being close to the current best params.

Reply:
The code functionality was clear... I was asking why the specific values of e5 and e-.1 ?

Reply:
Ah, yeah the specific choice is kind of arbitrary, but the idea was to get a range that went from a relatively small perturbation (`1e-5`) to a perturbation that moves towards the upper bound (`0.9`).